wgpu-core/src/device/mod.rs

```diff
@@ -634,9 +636,24 @@ impl<A: HalApi> Device<A> {
         adapter: &crate::instance::Adapter<A>,
         desc: &resource::TextureDescriptor,
     ) -> Result<resource::Texture<A>, resource::CreateTextureError> {
-        // Enforce COPY_DST, otherwise we wouldn't be able to initialize the texture.
-        let hal_usage =
-            conv::map_texture_usage(desc.usage, desc.format.into()) | hal::TextureUses::COPY_DST;
+        let format_desc = desc.format.describe();
+
+        // Depth volume textures can't be written to - depth forbids COPY_DST and volume textures can't be rendered to - therefore they aren't allowed.
+        if format_desc.sample_type == wgt::TextureSampleType::Depth
+            && desc.dimension == wgt::TextureDimension::D3
+        {
+            return Err(resource::CreateTextureError::CannotCreateDepthVolumeTexture(desc.format));
+        }
+
+        // Enforce having COPY_DST/DEPTH_STENCIL_WRIT/COLOR_TARGET otherwise we wouldn't be able to initialize the texture.
+        let hal_usage = conv::map_texture_usage(desc.usage, desc.format.into())
+            | if format_desc.sample_type == wgt::TextureSampleType::Depth {
+                hal::TextureUses::DEPTH_STENCIL_WRITE
+            } else if desc.sample_count > 1 {
+                hal::TextureUses::COLOR_TARGET
+            } else {
+                hal::TextureUses::COPY_DST
```

Comment:
I think we need to change this. We don't want to add COPY_DST usage if the user didn't request it, since it may slow down the other operations on the texture. Therefore, this should look like:
```rust
} else if desc.usage.contains(wgt::TextureUsage::COPY_DST) {
  hal::TextureUses::COPY_DST
} else {
  hal::TextureUses::COLOR_TARGET
}
```

I don't even think this needs to check for MSAA, since MSAA textures shouldn't support COPY_DST at all.

Reply:
agreed, we should always use COLOR_TARGET over COPY_DST, but in any case where a user creates a texture that can not get COLOR_TARGET, we still need to enforce COPY_DST.
This would happen either when user intends to write via STORAGE or by error (or ofc banking on us not doing zero-init and trying to read uninitialized memory ;))

Reply:
this now made the volume textures in the clear_texture tests being cleared by via render target, making all sort of bugs visible now. Not the first time this test saved me, feels like we need more tests that grind through all permutations of a thing :)